docs/hardware/toonv1.md

```diff
@@ -0,0 +1,245 @@
+* CPU: ARM926EJ-S
+* Speed: 400 MHz
+
+[Datasheet](./datasheets/NXP-MCIMX27LMOP4A.pdf)
+
+## Memory
+NANYA NT6DM32M32BC-T1
+* Size: 128 MiB
+* Type: DDR
+* Speed: 200MHz
+
+[Datasheet](./datasheets/NANYA-NT6DM32M32BC-T1.pdf)
+
+## Storage
+Samsung K9F1G08U0D
+* Size: 128 MiB
+* Type: NAND Flash
+
+[Datasheet](./datasheets/Samsung-K9F1G08U0D.pdf)
+
+### Partition table
+  #   | size    | label           | notes
+  --- | ---     | ---             | ---
+  0   | 1MiB    | `u-boot`        | Bootloader; not mapped in Linux
+  1   | 512KiB  | `u-boot-env`    | U-Boot environment variables
+  2   | 1536KiB | `splash-image`  | Boot splash screen
+  3   | 3MiB    | `kernel`        | Linux kernel
+  4   | 3MiB    | `kernel-backup` | Backup Linux kernel
+  5   | 119MiB  | `rootfs`        | Main root filesystem
+
+## Display
+Tianma TM070RDH03
+* Size: 7 inch
+* Resolution: 800x480 (WVGA), 134PPI, 16:9 aspect ratio
+* Type: TFT LCD
+* Touch: 4-wire resistive touch screen
+
+No datasheet found.
+
+## Interfaces
+
+### Ethernet
+* 10/100 Mbps Ethernet port
+* Embedded, on-SoC PHY (Freescale Fast Ethernet Controller)
+
+See the SoC's datasheet for more information.
+
+### WiFi
+AzureWave AW-NU168H
+* Mini PCIe module
+* Ralink RT5390 802.11b/g/n chip
+* Connected via internal USB2 interface (see USB2 section below).
+
+[Datasheet](./datasheets/AzureWave-AW-NU168H.pdf)
+
+### Z-wave
+Sigma designs ZW0301 Z-Wave controller
+
+[Datasheet](./datasheets/Sigma-Designs-ZW0301.pdf)
+
+The Z-Wave controller is connected to the SoC via UART2 (ttymxc2), which is
+used for communication with the Z-Wave controller. The Z-Wave controller also
+has a serial EEPROM connected to it, which is used for data storage.
+
+### Serial EEPROM
+Atmel `ATMLH402` (as marked on chip)
+* SPI serial EEPROM
+* 256 Kbit (32 KiB x 8 bits)
+* Address: spi0.0
+* Linux driver: `at25`
+
+This EEPROM is likely not supposed to be accessed directly, but rather through
+the Z-Wave controller's interface. However, the SPI bus is accessible from
+Linux, and the EEPROM can be accessed directly.
+
+[Datasheet](./datasheets/Fujitsu-MB85RS256TY.pdf) of a compatible part.
+
+### USB2
+Microchip USB3370 (x2)
+* Enhanced Single Supply Hi-Speed USB ULPI Transceiver (USB PHY)
+* Used to connect the USB port and the Wi-Fi module
+* Connected to the SoC's USB2 ULPI interface
+
+[Datasheet](./datasheets/Microchip-USB3370.pdf)
+
+
+### OpenTherm / power port
+Toon v1 has a 2-pin port on its back which is used to connect the Toon to a
+central heating system or boiler, as well as to provide power.
+
+The pins match up with a connector embedded in the wall mount.
+
+OpenTherm communications are done through UART1 (ttymxc1), at 4800,8n1 (4800
+baud(?), 8 data bits, no parity, 1 stop bit).
+
+#### Ketelmodule
+Commonly, the wall mount connector is connected to a Ketelmodule (boiler
+module). This module is a separate device, and allows the Toon to communicate
```

Comment:
(this note applies in many places) - I really like exactly one sentence per line, it makes diffs look so much better.